test/libsolidity/semanticTests/viaYul/comparison.sol

```diff
@@ -0,0 +1,74 @@
+contract C {
+    function f(address a) public pure returns (bool) {
+        return a == address(0);
+    }
+    function g() public pure returns (bool) {
+        return bytes3("abc") == bytes4("abc");
+    }
+    function lt(uint a, uint b) public pure returns (bool) {
+        return a < b;
+    }
+    function slt(int a, int b) public pure returns (bool) {
+        return a < b;
+    }
+    function lte(uint a, uint b) public pure returns (bool) {
+        return a <= b;
+    }
+    function slte(int a, int b) public pure returns (bool) {
+        return a <= b;
+    }
+    function gt(uint a, uint b) public pure returns (bool) {
+        return a > b;
+    }
+    function sgt(int a, int b) public pure returns (bool) {
+        return a > b;
+    }
+    function gte(uint a, uint b) public pure returns (bool) {
+        return a >= b;
+    }
+    function sgte(int a, int b) public pure returns (bool) {
+        return a >= b;
+    }
+    function eq(uint a, uint b) public pure returns (bool) {
+        return a == b;
+    }
+    function neq(uint a, uint b) public pure returns (bool) {
```

Comment:
What about tests for differently sized (u)ints, 8,16,32, etc?

Reply:
I think the type conversion PR tested those.

Reply:
It didn't.